### PR TITLE
Add value conversion tables

### PR DESCRIPTION
**Pull request by kdaun**
Adds value conversion tables to perform lazy computations of lookup tables for mapping from a uint16 value to a float in [`lower_bound`, `upper_bound`].
Owners of the value conversion tables are `SubmapController`, `PoseGraph2D`, `ActiveSubmaps` and `ProbabilityGridPointsProcessor`. For concurrency reasons, having only a single owner is not possible.
Follow up PR in `cartographer_ros`  is prepared.

### CODE REVIEW

File: cartographer/mapping/value_conversion_tables.cc [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+namespace cartographer {
+namespace mapping {
+namespace {
+
+// 0 is unknown, [1, 32767] maps to [lower_bound, upper_bound].
+float SlowValueToBoundedFloat(const uint16 value, const uint16 unknown_value,
+                              const float unknown_result,
+                              const float lower_bound,
+                              const float upper_bound) {
+  CHECK_GE(value, 0);
```
Comment (cschuet) [inline cartographer/mapping/value_conversion_tables.cc]: Not sure if a uint16 can be less than zero ;)

File: cartographer/mapping/value_conversion_tables.cc [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  CHECK_LE(value, 32767);
+  if (value == unknown_value) return unknown_result;
+  const float kScale = (upper_bound - lower_bound) / 32766.f;
+  return value * kScale + (lower_bound - kScale);
+}
+
+std::unique_ptr<std::vector<float>> PrecomputeValueToBoundedFloat(
+    const uint16 unknown_value, const float unknown_result,
+    const float lower_bound, const float upper_bound) {
+  auto result = common::make_unique<std::vector<float>>();
```
Comment (cschuet) [inline cartographer/mapping/value_conversion_tables.cc]: reserve space in vector.

File: cartographer/mapping/value_conversion_tables.cc [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+          value, unknown_value, unknown_result, lower_bound, upper_bound));
+    }
+  }
+  return result;
+}
+}  // namespace
+
+const std::vector<float>* ValueConversionTables::GetConversionTable(
+    float unknown_result, float lower_bound, float upper_bound) {
+  if (lower_bound == 0.f && upper_bound == 0.f) {
```
Comment (cschuet) [inline cartographer/mapping/value_conversion_tables.cc]: So an old pbstream would run into this if? Also pbstreams for old OccGrids or only TSDF?

File: cartographer/mapping/value_conversion_tables.cc (hunk 1 above)
Comment (kdaun) [reply cartographer/mapping/value_conversion_tables.cc]: True, I'll also remove it from 'probability_values.cc'.

File: cartographer/mapping/value_conversion_tables.cc (hunk 2 above)
Comment (kdaun) [reply cartographer/mapping/value_conversion_tables.cc]: Yes.

**Comment by kdaun** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user kdaun. Merge queue now has a length of 2.

**Pull request merged**